workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/ui/WorkflowTree.kt

```diff
@@ -53,33 +70,208 @@ internal fun DrawTree(
     val isExpanded = expandedNodes[node.id] == true
 
     DrawNode(
-      node,
-      previousNode,
-      isAffected,
-      isExpanded,
-      onNodeSelect,
-      onExpandToggle = { expandedNodes[node.id] = !expandedNodes[node.id]!! }
+      node = node,
+      nodePast = previousFrameNode,
+      isAffected = isAffected,
+      isExpanded = isExpanded,
+      onNodeSelect = onNodeSelect,
+      onExpandToggle = { expandedNodes[node.id] = !expandedNodes[node.id]!! },
+      storeNodeLocation = storeNodeLocation
     )
 
-    // Draws the node's children recursively.
     if (isExpanded) {
+      val (affectedChildren, unaffectedChildren) = node.children.values
+        .partition { affectedNodes.contains(it) }
+
+      UnaffectedChildrenGroup(
+        node = node,
+        children = unaffectedChildren,
+        previousFrameNode = previousFrameNode,
+        affectedNodes = affectedNodes,
+        expandedNodes = expandedNodes,
+        onNodeSelect = onNodeSelect,
+        storeNodeLocation = storeNodeLocation
+      )
+
+      AffectedChildrenGroup(
+        children = affectedChildren,
+        previousFrameNode = previousFrameNode,
+        affectedNodes = affectedNodes,
+        expandedNodes = expandedNodes,
+        onNodeSelect = onNodeSelect,
+        storeNodeLocation = storeNodeLocation
+      )
+    }
+  }
+}
+
+/**
+ * Draws the group of unaffected children, which can be open and closed to expand/collapse them.
+ *
+ * If an unaffected children also has other children, it cannot be opened since the this group
+ * treats all nodes as one entity. The right click for the whole group overrides the right click for
+ * the individual nodes.
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun UnaffectedChildrenGroup(
+  node: Node,
+  children: List<Node>,
+  previousFrameNode: Node?,
+  affectedNodes: Set<Node>,
+  expandedNodes: MutableMap<String, Boolean>,
+  onNodeSelect: (NodeUpdate) -> Unit,
+  storeNodeLocation: (Node, Offset) -> Unit
+) {
+  if (children.isEmpty()) return
+
+  val groupKey = "${node.id}_unaffected_group"
+  LaunchedEffect(Unit) {
```

Comment:
Nit: This doesn't need a coroutine, it would be slightly more efficient to use `DisposableEffect`.